tests/builder.rs

```diff
@@ -0,0 +1,84 @@
+extern crate alloc;
+use alloc::sync::Arc;
```

Comment:
Tests in the toplevel `tests/` folder always link `std`:

```suggestion
use std::sync::Arc;
```